tests/providers/openlineage/plugins/test_openlineage.py

```diff
@@ -131,7 +131,7 @@ def teardown_method(self):
             ),
         ],
     )
-    def test_plugin_disablements(self, mocks, expected):
+    def test_plugin_disablements(self, mocks, expected, require_running_tests_from_sources):
```

Comment:
@mobuchowski @kacpermuda -> This test stops working when running the test with Airflow 2.9.1, I am not sure if this is expected, and it likely comes from the way how plugins are initialized, I will need some help in making the test compatible with running it against 2.9.1.

Reply:
Unfortunately I'm not up to date with the changes being made to the plugin initialization, so I'll have to look into it, but I'll be happy to help with fixing those tests.

Reply:
OK. I think I fixed it by making the imports local to the tests.  When airflow is installed from package, just importing airflow package will initialize the plugins **before** patching done by the test. But by moving airflow imports to inside setup/teardown methods, we delay plugin initializiation to the setup method which is is run after patching of the os environment is done for the test. That seems to fix the problem.

Reply:
Hmm. Not really - seems that every time we re-run the test wiht 2.9.1 installed as package, it works. It stops working when I re-run it with `--with-db-init` switch.

Steps to reproduce:

1) Start breeze with airflow 2.9.1  installed from package:

```
breeze shell --use-packages-from-dist --package-format wheel --use-airflow-version 2.9.1 --airflow-constraints-reference constraints-2.9.1 --providers-skip-constraints --install-airflow-with-constraints --mount-sources test
```

2) pytest tests/providers/openlineage/plugins/test_openlineage.py -> errors
3) pytest tests/providers/openlineage/plugins/test_openlineage.py -> success
4) pytest tests/providers/openlineage/plugins/test_openlineage.py -> success 
5) pytest tests/providers/openlineage/plugins/test_openlineage.py --with-db-init -> errors
6) pytest tests/providers/openlineage/plugins/test_openlineage.py --with-db-init -> errors
7) pytest tests/providers/openlineage/plugins/test_openlineage.py -> success 

Very repeatable.

Reply:
> But by moving airflow imports to inside setup/teardown methods, we delay plugin initializiation to the setup method which is is run after patching of the os environment is done for the test. That seems to fix the problem.

Hmm... not sure how local imports and clearing `lru_cache` interacts... but I think tests shouldn't really rely on Airflow initialization. Not sure how to do it yet.

Reply:
Yep. It's rather strange behaviour - because it's not even local imports. Seems that running the tests ONCE (without resetting the DB) makes the test work fine afterwards (until we reset the DB again). So it looks like a side effect that relies on a database state **somewhere**

Reply:
OK @mobuchowski  @kacpermuda . I know **why** it happens. 

I do not yet know the solution, but the problem is that when airflow is installed from the package, plugins get initialized when `airflow db init` is run before tests start and they are not re-initialized later so the `is_disabled` method is not called in the tests at all - they all use hooks that get registered during plugin initialization when `db init` is called. Now ... We should likely find a way to unload plugins in this case.